openquake/calculators/event_based.py

```diff
@@ -406,6 +440,8 @@ def execute(self):
         scenario = 'scenario' in oq.calculation_mode
         proxies = [RuptureProxy(rec, scenario)
                    for rec in dstore['ruptures'][:]]
+        if "station_data" in oq.inputs:
+            proxies = proxies[0:1]
```

Comment:
Care to add a comment here? If would have expected an error if there is more than one rupture, not silently ignoring ruptures.

Reply:
What's happening here is that for a regular scenario gmf calculation, if the user requests 100 gmfs, the engine copies the same rupture 100 times before parallelizing by rupture. However, for a conditioned gmf calculation for a scenario, the conditioning process is computationally quite expensive, so we want to avoid repeating it 100 times. Suggestions for a cleaner way of handling this case?

Reply:
Added a comment in the code